### PR TITLE
[SDK] add Dart to path

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

The windows implementation of placing a dart executable on the PATH. This moves all of the shared logic into internal/shared.bat and adds a CALL to invoke this from both flutter.bat and dart.bat. 

To keep the code simple, the bootstrap process is identical whether calling into dart or flutter.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
This wouldn't be landed until we're ready with the bash implementation too

File: bin/dart.bat [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+SETLOCAL ENABLEDELAYEDEXPANSION
+
+FOR %%i IN ("%~dp0..") DO SET FLUTTER_ROOT=%%~fi
+
+REM Include shared scripts in shared.bat
+SET shared_bin=%FLUTTER_ROOT%/bin/internal/shared.bat
+CALL "%shared_bin%"
+
+SET flutter_tools_dir=%FLUTTER_ROOT%\packages\flutter_tools
```
Comment (goderbauer) [inline bin/dart.bat:22]: nit: some of the variables set here seem unused? e.g. snapshot_path and engine_stamp

File: bin/dart.bat [hunk 2]
```diff
@@ -0,0 +1,46 @@
+@ECHO off
+REM Copyright 2014 The Flutter Authors. All rights reserved.
+REM Use of this source code is governed by a BSD-style license that can be
+REM found in the LICENSE file.
+
+REM ---------------------------------- NOTE ----------------------------------
+REM
+REM Please keep the logic in this file consistent with the logic in the
+REM `flutter` script in the same directory to ensure that Flutter continues to
```
Comment (goderbauer) [inline bin/dart.bat:9]: Does this comment need updating (i.e. is the linux side also getting restructured?)

File: bin/internal/shared.bat [hunk 3]
```diff
@@ -0,0 +1,162 @@
+@ECHO off
+REM Copyright 2014 The Flutter Authors. All rights reserved.
+REM Use of this source code is governed by a BSD-style license that can be
+REM found in the LICENSE file.
+
+REM ---------------------------------- NOTE ----------------------------------
+REM
+REM Please keep the logic in this file consistent with the logic in the
+REM `flutter` script in the same directory to ensure that Flutter continues to
```
Comment (goderbauer) [inline bin/internal/shared.bat:9]: this may need updating (see other comment)

**Review by goderbauer — COMMENTED** [review]
Looks fine.

**Review by zanderso — APPROVED** [review]

File: bin/dart.bat (hunk 2 above)
Comment (comm1x) [reply bin/dart.bat:9]: Ok, that's fine

**Comment by jonahwilliams** [comment]
Updated to remove unused variables, and reference update pwsh location and disable-dartdev for flutter.

File: bin/internal/shared.bat (hunk 3 above)
Comment (jonahwilliams) [reply bin/internal/shared.bat]: Done

File: bin/dart.bat (hunk 2 above)
Comment (jonahwilliams) [reply bin/dart.bat]: Done

File: bin/dart.bat (hunk 1 above)
Comment (jonahwilliams) [reply bin/dart.bat]: Removed

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
This was pulled into @christopherfujino 's change to land in one go

**Pull request closed without merging**